### PR TITLE
Option to generate source maps when bundling

### PR DESCRIPTION
**Pull request by nevir**
This is currently untested :S  (I'm waiting for the 0.13 changes to land, and I'll bump my project to that & test then)

### CODE REVIEW

File: lib/createBundle.js [hunk 1]
```diff
@@ -15,23 +15,38 @@ const fetch = require('./fetch');
...
 module.exports = function createBundle(server, options) {
-  const bundleUrl = url.format({
+  let promise = download(buildUrl(server, options, 'bundle'), options.targetPath);
+  if (options.sourceMap) {
+    promise = promise.then(() => {
+      download(buildUrl(server, options, 'map'), `${options.targetPath}.map`);
+    });
+  }
+
+  return promise;
```
Comment (elliottsj) [inline lib/createBundle.js]: Can you instead construct two promises (for bundle & map) and return `Promise.all()`? Something along the lines of

``` js
const bundlePromise =
  fetchBundle(server, options)
    .then(content => writeFile(options.targetPath, content));

const mapPromise =
  !options.sourceMap ?
    Promise.resolve() :
    fetchMap(server, options)
      .then(content => writeFile(`${options.targetPath}.map`, content));

return Promise.all([bundlePromise, mapPromise]);
```

**Comment by nevir** [comment]
Should be good to go now

File: lib/createBundle.js [hunk 2]
```diff
@@ -15,23 +15,37 @@ const fetch = require('./fetch');
...
       minify: options.minify,
     },
-  });
-  return fetch(bundleUrl).then(content =>
-    mkdirp.mkdirpAsync(path.dirname(options.targetPath)).then(() =>
-      fs.writeFileAsync(options.targetPath, content)
+  })
+}
+
+function download(url, taretPath) {
```
Comment (elliottsj) [inline lib/createBundle.js]: `taretPath` -> `targetPath`

**Comment by elliottsj** [comment]
Just one last typo, then will merge :+1:

File: lib/createBundle.js (hunk 2 above)
Comment (nevir) [inline lib/createBundle.js]: Cuuuurses!

**Comment by nevir** [comment]
done :P

**Comment by elliottsj** [comment]
Thanks

**Pull request merged**